core/templates/components/state-editor/state-skill-editor/state-skill-editor.component.ts

```diff
@@ -55,16 +59,35 @@ export class StateSkillEditorComponent implements OnInit {
     private windowDimensionsService: WindowDimensionsService,
     private stateLinkedSkillIdService: StateLinkedSkillIdService,
     private urlInterpolationService: UrlInterpolationService,
-    private ngbModal: NgbModal
+    private ngbModal: NgbModal,
+    private skillBackendApiService: SkillBackendApiService,
+    private userService: UserService
   ) {}
 
   ngOnInit(): void {
     this.skillEditorIsShown = (!this.windowDimensionsService.isWindowNarrow());
-    this.topicsAndSkillsDashboardBackendApiService.fetchDashboardDataAsync()
-      .then((response: TopicsAndSkillDashboardData) => {
-        this.categorizedSkills = response.categorizedSkillsDict;
-        this.untriagedSkillSummaries = response.untriagedSkillSummaries;
+    this.userService.getUserInfoAsync()
+      .then((userInfo) => {
+        this.isEditableByUser = (
+          userInfo.isLoggedIn() &&
+          (userInfo.isCurriculumAdmin() || userInfo.isTopicManager())
+        );
+        if (this.isEditableByUser) {
+          this.topicsAndSkillsDashboardBackendApiService
+            .fetchDashboardDataAsync()
+            .then((response: TopicsAndSkillDashboardData) => {
+              this.categorizedSkills = response.categorizedSkillsDict;
+              this.untriagedSkillSummaries = response.untriagedSkillSummaries;
+            });
+        }
       });
+    if (this.stateLinkedSkillIdService.displayed) {
+      this.skillBackendApiService.fetchSkillAsync(
+        this.stateLinkedSkillIdService.displayed
+      ).then((skill) => {
+        this.skillName = skill.skill._description;
```

Comment:
Can we use the relevant getter in the skill object instead of using a private variable?

Reply:
Done